File: ui/predictball/src/app/models/competition.ts
export interface Season {
    id: number;
    startDate: string;
    endDate: string;
    currentMatchday: number;
    winner: any | null;
    isRetired?: boolean;
    isFinished?: boolean;
}

export interface Competition {
    id: number;
    name: string;
    code: string;
    type: string;
    emblem: string;
    currentSeason?: Season;
    seasons?: Season[];
}

File: api/models/football-data/competition.go
package footballdata

type Competition struct {
	ID            int      `json:"id"`
	Name          string   `json:"name"`
	Code          string   `json:"code"`
	Type          string   `json:"type"`
	Emblem        string   `json:"emblem"`
	CurrentSeason Season   `json:"currentSeason"`
	Seasons       []Season `json:"seasons,omitempty"`
	Area          Area     `json:"area,omitempty"`
}
